gh-badges/lib/index.js

```diff
@@ -17,6 +19,56 @@ class BadgeFactory {
     }
   }
 
+  _validate(format) {
+    if (!('text' in format)) {
+      throw new ValidationError('Field `text` is required')
+    }
+
+    if (
+      !Array.isArray(format.text) ||
+      format.text.length !== 2 ||
+      typeof format.text[0] !== 'string' ||
+      typeof format.text[1] !== 'string'
+    ) {
+      throw new ValidationError('Field `text` must be an array of 2 strings')
+    }
+
+    const stringFields = [
+      'labelColor',
+      'color',
+      'colorA',
+      'colorscheme',
+      'colorB',
```

Comment:
Would it be worth another check to print a warning message when any of these deprecated fields are used, similar to the constructor check on `option`?

Reply:
One of the things on the v3 roadmap is to remove the params we deprecated in 2.2.0. I was aiming to keep each of those points a separate PR, but maybe it makes sense to just remove the deprecated params in this PR rather as well than add validation for them here and then remove them later.